test/js/pages.test.js

```diff
@@ -0,0 +1,165 @@
+      tags: ['tag3', 'tag3'],
+      author: 'erica',
+      end: '2020-01-09T04:10:17.000Z',
+      renderData: { foo: 'bar' },
+    },
+    templateContent: '<h1>This is my title</h1>\n\n<p>This is content…',
+  },
+  {
+    inputPath: './test4.md',
+    fileSlug: 'test4',
+    outputPath: './_site/test4/index.html',
+    url: '/test4/',
+    date: '2019-01-09T04:10:17.000Z',
+    data: {
+      title: 'Event Title',
+      tags: ['tag3', 'tag3'],
+      author: 'erica',
+      end: '2020-01-09T04:10:17.000Z',
+      foo: { foo: 'bar' },
+      draft: true,
+      events: [
+        {
+          foo: 'bar',
+          date: '2018-01-09T04:10:17.000Z',
+          end: '2018-01-10T04:10:17.000Z',
+        },
+        {
+          bar: 'baz',
+          date: '2018-02-09T04:10:17.000Z',
+          end: '2018-01-10T04:10:17.000Z',
+        },
+      ],
+    },
+    templateContent: '<h1>This is my title</h1>\n\n<p>This is content…',
+  },
+];
+describe('page filters', () => {
+  test('isPublic', () => {
+    expect(isPublic(collection[0])).toBeTruthy();
+    expect(isPublic(collection[1])).toBeFalsy();
+  });
+
+  test('isCurrent', () => {
+    expect(isCurrent(collection[0])).toBeTruthy();
+    expect(isCurrent(collection[2])).toBeFalsy();
+  });
+
+  test('getPublic', () => {
+    const publicPages = [collection[0], collection[2]];
+
+    expect(getPublic(collection)).toEqual(publicPages);
+  });
+
+  test('hasData', () => {
+    // this keeps returning undefined...
+    hasData(collection, 'data.slug', 'news');
+  });
+
+  test('getData', () => {
+    // this keeps returning undefined...
+    getData(collection, 'author', 'erica');
+  });
+
+  test('findData', () => {
+    // this keeps returning undefined...
+    findData(collection, 'author', 'erica');
+  });
+
+  test('getPage', () => {
+    // this keeps returning undefined...
+    getPage(collection, '/test2/', 'foo', 'bar');
+  });
+
+  test('findPage', () => {
+    // this keeps returning undefined...
+    findPage(collection, '/test2/', 'foo', 'bar');
+  });
+
+  test('render', () => {
+    expect(render(collection[2], 'foo')).toBe('bar');
+    expect(render(collection[3], 'foo')).toEqual({ foo: 'bar' });
+  });
+
+  test('pageYears', () => {
+    const testPage = pageYears(collection)[0];
+    const pageKeys = Object.keys(testPage);
+
+    expect(pageKeys).toHaveProperty('sort');
+    // This is somehow failing but its in keys when printed //
+    expect(pageKeys).toHaveProperty('year');
+  });
+
+  describe('byYear', () => {
+    test('returns empty if no collection', () => {
+      const expected = [];
+
+      expect(byYear([])).toEqual(expected);
+    });
+    test('returns pages grouped by year', () => {
+      byYear(collection);
+    });
+  });
+
+  describe('pageType', () => {
+    test('Return one of several resource "types" from page tags', () => {
+      const tags = ['Workshops', 'Podcasts', 'foo', 'bar'];
+      expect(pageType(tags)).toEqual(['Workshops', 'Podcasts']);
```

Comment:
This should just match the _first_ occurrence, not all matching ones.

Reply:
OK makes sense.  In `pageType` I feel like it would make more sense to return an empty string instead of false if the tag isn't included in the types list.